controllers/awscluster_controller.go

```diff
@@ -160,7 +160,7 @@ func reconcileNormal(clusterScope *scope.ClusterScope) (reconcile.Result, error)
 		return reconcile.Result{}, errors.Wrapf(err, "failed to reconcile load balancers for AWSCluster %s/%s", awsCluster.Namespace, awsCluster.Name)
 	}
 
-	if awsCluster.Status.Network.APIServerELB.DNSName == "" {
+	if awsCluster.Status.Network.APIServerELB.DNSName == "" || len(awsCluster.Status.Network.APIServerELB.Listeners) < 1 {
```

Comment:
Do we actually need to get the value from the Listener?

I'm wondering if we can just use the same value that `s.scope.APIServerPort()` is derived from when creating the Load Balancer?

Reply:
Not sure if we should assume that the APIServerPort is the LoadBalancerPort, given that users might want to change the port the ELB listens on in the future

Reply:
I would expect that to be driven by some type of config that we could also reference. I wouldn't expect users to modify the Listener out of band and for us to need to introspect the value to know what it is.

Reply:
I'd prefer to use what's returned by the describe call. I don't want to give users the false hope that changing the APIServerPort on the Cluster object, changes also the ELB port

Reply:
We also have to remember that adding additional calls counts against quota that we are already hitting in some cases. If we are reconciling the load balancer declaratively then we can assume that the value we are setting in the APIEndpoints will eventually be honored through reconciliation.

Reply:
Is there something we can pull into an issue here to discuss? What should we do in the interim?

Master cannot create clusters in current state.

Reply:
Don't we hardcode the port today? If so, we should just hardcode it here as well and add a TODO comment to fix it up when we have a way for users to specify it

Reply:
spec.ClusterNetwork.APIServerPort seems to be honored 

```
// APIServerPort returns the APIServerPort to use when creating the load balancer.
func (s *ClusterScope) APIServerPort() int64 {
	if s.Cluster.Spec.ClusterNetwork != nil && s.Cluster.Spec.ClusterNetwork.APIServerPort != nil {
		return int64(*s.Cluster.Spec.ClusterNetwork.APIServerPort)
	}
	return 6443
}
```

Reply:
I was referring to https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/pkg/cloud/services/elb/loadbalancer.go#L174

Reply:
The APIEndpoint port should reference the Load Balancer port, right, not the instance port? The instance port appears to be hardcoded but the listener port is set to whatever API server port is in the Spec.

Sorry I think I'm confused lol.

Reply:
I'll undo the change getting it from the describe output and use the scope's port for now, with a TODO.

Reply:
I'd prefer to hardcode, like we do in the other place I linked above, otherwise if you change the APIServer port, the listener won't change but only the backend port will, which is going to result in a bug